provisioning/miniprov/test/test_cdf.py

```diff
@@ -363,21 +367,19 @@ def ret_motr_mdvalues(value: str) -> Any:
 
         motr_store._raw_get = Mock(side_effect=ret_motr_mdvalues)
 
-        cdf = CdfGenerator(provider=store,
-                           motr_provider=motr_store)
+        cdf = CdfGenerator(provider=store)
         cdf.utils = utils
         ret = cdf._create_node_descriptions()
 
         self.assertIsInstance(ret, list)
-        self.assertEqual(1, len(ret))
+        self.assertEqual(2, len(ret))
```

Comment:
Why we are checking if len is 2 even if we provided only 1 node 
`'cluster>storage_set[0]>nodes': ['MACH_ID'],`

Reply:
Yeah,

```
    def _create_node_descriptions(self) -> List[NodeDesc]:
        nodes: List[NodeDesc] = []
        conf = self.provider
        # Skipping for controller and HA pod
        machine_ids = conf.get_machine_ids_for_service(
            Const.SERVICE_MOTR_IO.value)

        machine_ids.extend(conf.get_machine_ids_for_service(
            Const.SERVICE_S3_SERVER.value))

        for machine_id in machine_ids:
            nodes.append(self._create_node(machine_id))
        return nodes
```
If all the components are mentioned on the same node, then above code might return a list with 2 items with duplicate machine ids. I think the list needs to be de-depulicated.

Reply:
@SwapnilGaonkar7 fixed the duplicate entries in machine_ids list.